casbin/config/config.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#ifdef CASBIN_EXPORTS
+#define CONFIG_API __declspec(dllexport)
+#else
+#define CONFIG_API __declspec(dllimport)
+#endif
+
+#include<string>
+#include<mutex>
+#include<map>
+#include<sstream>
+#include<vector>
+#include"../errors/exceptions.h"
+
+
+using namespace std;
+
+extern const string DEFAULT_SECTION;
+extern const string DEFAULT_COMMENT;
+extern const string DEFAULT_COMMENT_SEM;
+extern const string DEFAULT_MULTI_LINE_SEPARATOR;
+
+
+
+class CONFIG_API Config {
+public:
+	map<string, map<string, string>> data;
+	mutex m;
+	Config();
+	static Config* NewConfigFromFile(Error& err,const string& path);
+	static Config* NewConfigFromText(Error& err, const string& text);
+	Error parseFile(const string& fname);

```

Comment:
throw an error is better than return an error, because you can use catch in c++

Reply:
I don't use throw because it is banned in some style.But if you don't mind it,I will use throw exception instead of Error class.

Reply:
I think hsluoyz didn't ban throw :smile:

Reply:
I will use throw exception instead of returning error in go version,and catch exception instead of   getting error.

Reply:
If it get an error and return it,I will do nothing.

Reply:
When should I catch a exception?Should I catch it when it happen or only throw it without catch.

Reply:
There is two idea:
1.When the exception happens,I throw it,and lets the user to catch it or lead to a crash.
2.When the exception happens,I catch it in casbin and log it,the program still run.

Reply:
You don't need to catch error in the library, just do the throw, it's the user's responsibility to catch an error.